crates/pixi_cli/src/build.rs

```diff
@@ -146,6 +169,8 @@ pub async fn execute(args: Args) -> miette::Result<()> {
 
     // Build the individual packages
     for package in packages {
+        let pinned_build_source = lock_file.pinned_build_source(&package);
```

Comment:
How does this function work? Which environment does it get this information from?